lib/rsa.js

```diff
@@ -887,6 +889,29 @@ pki.rsa.generateKeyPair = function(bits, e, options, callback) {
   // key generation code if available and if parameters are acceptable
   if(!forge.options.usePureJavaScript && callback &&
     bits >= 256 && bits <= 16384 && (e === 0x10001 || e === 3)) {
+    if(_detectNodeCrypto('generateKeyPair')) {
+      return _crypto.generateKeyPair('rsa', {
+        modulusLength: bits,
+        publicExponent: e,
+        publicKeyEncoding: {
+          type: 'pkcs1',
+          format: 'pem'
+        },
+        privateKeyEncoding: {
+          type: 'pkcs8',
+          format: 'pem'
+        }
+      }, function(err, pub, priv) {
+        if (err) {
+          return callback(err);
+        }
+        var privateKey = pki.privateKeyFromPem(priv);
+        callback(null, {
+          privateKey: privateKey,
+          publicKey: pki.setRsaPublicKey(privateKey.n, privateKey.e)
```

Comment:
Thanks again, can `pub` be used directly somehow here?  I'm comparing this implementation to another recent bit of dev code.  I notice that `publicKeyEncoding.type` is `spki`.  Is that relevant here?

https://github.com/digitalbazaar/crypto-ld/blob/signer/lib/ld-key-pair.js#L357-L376

Reply:
Does `pki.publicKeyFromPem` do the right thing? https://github.com/digitalbazaar/forge/blob/26bf82aed3b9814ab032cb88244b8176d6653bb1/lib/x509.js#L751-L752

Reply:
Tests pass.

Guess I should've grepped the whole project for `publicKeyFromPem`. Just assumed it didn't exist because it wasn't next to `privateKeyFromPem` in `pki.js`.

Will update.

Reply:
Was this going to be updated to spki?

Reply:
Sorry, I don’t understand the question

Reply:
These resolved things are confusing me, i'll comment in the latest code.